src/main/java/org/jabref/gui/integrity/IntegrityCheckDialogViewModel.java

```diff
@@ -1,22 +1,150 @@
 package org.jabref.gui.integrity;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
+import javafx.collections.ObservableSet;
 
 import org.jabref.gui.AbstractViewModel;
 import org.jabref.logic.integrity.IntegrityMessage;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
 
 public class IntegrityCheckDialogViewModel extends AbstractViewModel {
 
     private final ObservableList<IntegrityMessage> messages;
+    private final ObservableSet<Field> entryTypes;
 
     public IntegrityCheckDialogViewModel(List<IntegrityMessage> messages) {
         this.messages = FXCollections.observableArrayList(messages);
+
+        Set<Field> types = messages.stream()
+                                    .map(IntegrityMessage::field)
+                                    .collect(Collectors.toSet());
+        this.entryTypes = FXCollections.observableSet(types);
     }
 
     public ObservableList<IntegrityMessage> getMessages() {
         return messages;
     }
+
+    public Set<Field> getEntryTypes() {
+        return entryTypes;
+    }
+
+    public void removeFromEntryTypes(String entry) {
+        entryTypes.remove(entry);
+    }
+
+    public void fix(IntegrityIssue issue, IntegrityMessage message) {
+        switch (issue) {
+            case CAPITAL_LETTER_ARE_NOT_MASKED_USING_CURLY_BRACKETS:
+                maskTitle(message);
+                break;
+            case BIBTEX_FIELD_ONLY_KEY, BIBTEX_FIELD_ONLY_CROSS_REF:
+                removeField(message, issue.getField());
+                break;
+            case INCORRECT_FORMAT_DATE:
+                correctDateFormat(message);
+                break;
+            case NO_INTEGER_AS_VALUE_FOR_EDITION_ALLOWED:
+                removeNonIntegerEdition(message);
+                break;
+            case SHOULD_CONTAIN_AN_INTEGER_OR_A_LITERAL:
+                ensureValidEdition(message);
+                break;
+            case SHOULD_HAVE_THE_FIRST_LETTER_CAPITALIZED:
+                capitalizeFirstLetter(message, StandardField.EDITION);
+                break;
+            case REFERENCED_CITATION_KEY_DOES_NOT_EXIST:
+                handleMissingCitationKey(message);
+                break;
+            case NON_ASCII_ENCODED_CHARACTER_FOUND:
+                replaceNonASCIICharacters(message, StandardField.ABSTRACT);
+                break;
+            case SHOULD_CONTAIN_A_VALID_PAGE_NUMBER_RANGE:
+                formatPageNumberRange(message);
+                break;
+            default:
+                break;
+        }
+    }
+
+    public void maskTitle(IntegrityMessage message) {
```

Comment:
I would like to see some tests for all these fix functions. Should be pretty easy to do